### PR TITLE
metrics: Add metrics about dropped messages in FilterStep

### PR DESCRIPTION
**Pull request by ayirr7**
For https://github.com/getsentry/arroyo/issues/257 

It seemed to me that we should emit this metric in both scenarios, whether a CommitPolicy was passed in or not. Might have missed something though.

Also, if there are other metrics that would be helpful to add in `FilterStep` strategy, I can include them in this PR.

### CODE REVIEW

File: arroyo/processing/strategies/filter.py [hunk 1]
```diff
@@ -80,7 +83,14 @@ def submit(
...
             self.__next_step.submit(message)
         elif self.__commit_policy_state is not None:
+            # first scenario, where a CommitPolicy is passed in
+            # and messages are dropped
             self.__uncommitted_offsets.update(message.committable)
+            self.__metrics.increment("arroyo.strategies.filter_step.dropped_messages")
+        elif self.__commit_policy_state is None:
+            # second scenario, where a CommitPolicy is not passed in
+            # and messages are dropped
+            self.__metrics.increment("arroyo.strategies.filter_step.dropped_messages")
```
Comment (lynnagara) [inline arroyo/processing/strategies/filter.py:93]: nit: The 2 elif statements make it harder to parse
```suggestion
        else:
            self.__metrics.increment("arroyo.strategies.filter_step.dropped_messages")
            if self.__commit_policy_state is not None:
                self.__uncommitted_offsets.update(message.committable)
```

File: arroyo/processing/strategies/filter.py [hunk 2]
```diff
@@ -80,7 +83,14 @@ def submit(
                 self.__uncommitted_offsets.pop(partition, None)
             self.__next_step.submit(message)
         elif self.__commit_policy_state is not None:
+            # first scenario, where a CommitPolicy is passed in
+            # and messages are dropped
             self.__uncommitted_offsets.update(message.committable)
+            self.__metrics.increment("arroyo.strategies.filter_step.dropped_messages")
```
Comment (lynnagara) [inline arroyo/processing/strategies/filter.py:89]: Can we make this:
```suggestion
            self.__metrics.increment("arroyo.strategies.filter.dropped_messages")
```
It kinda bugs me that a couple of the steps are suffixed with "step" and I think we should standardise it.

Comment (ayirr7) [reply arroyo/processing/strategies/filter.py]: Done

File: arroyo/processing/strategies/filter.py (hunk 1 above)
Comment (ayirr7) [reply arroyo/processing/strategies/filter.py]: Yep, makes sense

**Review by lynnagara — APPROVED** [review]
Thanks!

**Pull request merged**